---
title: "MFuzz_RNASeq_w0w1w6"
author: "Bruna Oriol"
date: "29/7/2020"
output: html_document
---

# MFuzz profiles DEG RNA Seq - w0, w1, w6 

MFuzz R package applies soft clustering to identify groups of genes with similar expression profiles along the time.

```{r}
rm(list=ls())
# Load libraries
library(Mfuzz)
```
 
## 1. Read normalized gene expression

```{r}
log2norm.counts <- read.table(file = "../RNASeq/Log2NormalizedCounts.txt" )
log2norm.counts <- log2norm.counts[, grep("w0|w1|w6", colnames(log2norm.counts))]
```


## 2. Read DEG: w1 vs w0 / w6 vs w0 / w6 vs w1

```{r}

DEG_w6w0 <- read.table("../RNASeq/DEGweeks/LimmaBlock w6vsw0.txt")
DEG_w6w0 <- subset(DEG_w6w0, DEG_w6w0$P.Value < 0.05 )
DEG_w1w0 <- read.table("../RNASeq/DEGweeks/LimmaBlock w1vsw0.txt")
DEG_w1w0 <- subset(DEG_w1w0, DEG_w1w0$P.Value < 0.05 )

Upw1w0 <- length(which(DEG_w1w0$logFC>0)) #673
Downw1w0 <- length(which(DEG_w1w0$logFC<0)) #524

Upw6w0 <- length(which(DEG_w6w0$logFC>0)) #1404
Downw6w0 <- length(which(DEG_w6w0$logFC<0)) #1561

DEG_w6w1 <- read.table("../RNASeq/DEGweeks/LimmaBlock w6vsw1.txt")
DEG_w6w1 <- subset(DEG_w6w1, DEG_w6w1$P.Value < 0.05 )


```

## 3. Take all the genes that are differentially expressed at some point

```{r}

DEGjoint <-  Reduce(unique,list(rownames(DEG_w6w0), rownames(DEG_w1w0), rownames(DEG_w6w1)))
DEGjoint.counts <- log2norm.counts[DEGjoint,]
Biotype <- sapply(strsplit(rownames(DEGjoint.counts), ","), function(x)x[3])
df.rownames <- data.frame(rownames(DEGjoint.counts), Biotype, stringsAsFactors = F)
rownames(df.rownames) <- df.rownames$rownames.DEGjoint.counts.

names.sel <- df.rownames[grep("protein_coding|^IG_|^TR_", df.rownames$Biotype),]
write.table(DEGjoint.counts, file = "../RNASeq/MFuzzProfiles_new/DEGw0w1w6joint.txt")
```

## 4. Now we need to take the mean of each of the genes in each timepoint.

```{r}
w0.counts <- DEGjoint.counts[, grep("w0", colnames(DEGjoint.counts))]
w0 <- apply(w0.counts, 1, median, drop = TRUE)
w1.counts <- DEGjoint.counts[, grep("w1", colnames(DEGjoint.counts))]
w1 <- apply(w1.counts, 1, median, drop = TRUE)
w6.counts <- DEGjoint.counts[, grep("w6", colnames(DEGjoint.counts))]
w6 <- apply(w6.counts, 1, median, drop = TRUE)
```
 

## 5. Construct a new matrix with weeks in columns (w0, w1 and w6) and genes in rows
1. Construct a matrix with the means calculated before: Genes in rows and weeks in columns.
2. Convert the matrix into an ExpressionSet
3. Standdardise the matrix: standardized value = (original value of mean) / standard deviation. It's important because
here we are not interested in genes with the same expression level but with the same expression profile.

```{r}
counts.matrix <- matrix(c(w0,w1,w6), ncol = 3, nrow = length(w0))
colnames(counts.matrix) <- c("w0", "w1", "w6")
#rownames(counts.matrix) <- sapply(strsplit(rownames(w0.counts), ","), function(x)x[1])
rownames(counts.matrix) <- rownames(w0.counts)

Eset.counts <-new("ExpressionSet", exprs=as.matrix(counts.matrix))

counts.matrix.s <- standardise(Eset.counts)
counts.matrix.s <- counts.matrix.s[complete.cases(counts.matrix.s@assayData$exprs),]
hist(counts.matrix.s@assayData$exprs)
```


##5. We need to estimate the fuzzier parameter. 
To do that, we apply the function mestimate().[Article Method] (https://academic.oup.com/bioinformatics/article/26/22/2841/227572)

```{r, mestimate}
m.est <- mestimate(counts.matrix.s)
m.est
```

# 6. Number of Clusters Determination
 We use the function Dmin() that calculates the minimum centroid distance between clusters for different clister numbes. 
```{r, Dmin}
Dmin(counts.matrix.s, m=m.est, crange=seq(3,10,1)) 

```

#7. MFuzz Plot

We will see the different ptofiles along the time.
We will select a minimal membership of 0.1 threshold to select the genes in each of the clusters. 

```{r, Make the plots for the clusters}
set.seed(42)

cl7 <- mfuzz(counts.matrix.s, c = 7, m = m.est)

pdf("../RNASeq/MFuzzProfiles_new/GeneProfiles_Allgenes_cl7.pdf")
mfuzz.plot(counts.matrix.s, cl = cl7, time.labels = c("w0", "w1", "w6"), mfrow = c(2,2), new.window = F, min.mem = 0.1)
dev.off()

```

```{r}


pdf("../RNASeq/MFuzzProfiles_new/GeneProfiles_Allgenes_cl7_cl1.pdf", width = 4, height = 4)
par(oma=c(0,0,0,1.2)) 
mfuzz.plot2(counts.matrix.s, cl = cl7, time.labels = c("Bsl", "Vacc", "Vacc+RMD"), mfrow = c(1,1), new.window = F, min.mem = 0.1, cex.axis = 1.4, x11=F, single =1, xaxt = "n", col.lab="transparent", col.main = "transparent")
axis(side = 1,labels = FALSE)
title(main = "Cluster 1\n (n = 656)", col.main ="black")
text(x = 1:3,
     ## Move labels to just below bottom of chart.
     y = par("usr")[3] - 0.45,
     ## Use names from the data list.
     labels = c("Bsl", "Vacc", "Vacc+RMD"),
     ## Change the clipping region.
     xpd = NA,
     ## Rotate the labels by 35 degrees.
     srt = 35,
     ## Adjust the labels to almost 100% right-justified.
     adj=0.4,
     ## Increase label size.
     cex = 1.4)

dev.off()

pdf("../RNASeq/MFuzzProfiles_new/GeneProfiles_Allgenes_cl7_cl2.pdf", width = 4, height = 4)
par(oma=c(0,0,0,1.2)) 
mfuzz.plot2(counts.matrix.s, cl = cl7, time.labels = c("Bsl", "Vacc", "Vacc+RMD"), mfrow = c(1,1), new.window = F, min.mem = 0.1, cex.axis = 1.4, x11=F, single =2, xaxt = "n", col.lab="transparent", col.main = "transparent")
axis(side = 1,labels = FALSE)
title(main = "Cluster 5\n (n = 250)", col.main ="black")
text(x = 1:3,
     ## Move labels to just below bottom of chart.
     y = par("usr")[3] - 0.45,
     ## Use names from the data list.
     labels = c("Bsl", "Vacc", "Vacc+RMD"),
     ## Change the clipping region.
     xpd = NA,
     ## Rotate the labels by 35 degrees.
     srt = 35,
     ## Adjust the labels to almost 100% right-justified.
     adj=0.4,
     ## Increase label size.
     cex = 1.4)

dev.off()

pdf("../RNASeq/MFuzzProfiles_new/GeneProfiles_Allgenes_cl7_cl3.pdf", width = 4, height = 4)
par(oma=c(0,0,0,1.2)) 
mfuzz.plot2(counts.matrix.s, cl = cl7, time.labels = c("Bsl", "Vacc", "Vacc+RMD"), mfrow = c(1,1), new.window = F, min.mem = 0.1, cex.axis = 1.4, x11=F, single =3, xaxt = "n", col.lab="transparent", col.main = "transparent")
axis(side = 1,labels = FALSE)
title(main = "Cluster 6\n (n = 206)", col.main ="black")
text(x = 1:3,
     ## Move labels to just below bottom of chart.
     y = par("usr")[3] - 0.45,
     ## Use names from the data list.
     labels = c("Bsl", "Vacc", "Vacc+RMD"),
     ## Change the clipping region.
     xpd = NA,
     ## Rotate the labels by 35 degrees.
     srt = 35,
     ## Adjust the labels to almost 100% right-justified.
     adj=0.4,
     ## Increase label size.
     cex = 1.4)

dev.off()

pdf("../RNASeq/MFuzzProfiles_new/GeneProfiles_Allgenes_cl7_cl4.pdf", width = 4, height = 4)
par(oma=c(0,0,0,1.2)) 
mfuzz.plot2(counts.matrix.s, cl = cl7, time.labels = c("Bsl", "Vacc", "Vacc+RMD"), mfrow = c(1,1), new.window = F, min.mem = 0.1, cex.axis = 1.4, x11=F, single =4, xaxt = "n", col.lab="transparent", col.main = "transparent")
axis(side = 1,labels = FALSE)
title(main = "Cluster 3\n (n = 324)", col.main ="black")
text(x = 1:3,
     ## Move labels to just below bottom of chart.
     y = par("usr")[3] - 0.45,
     ## Use names from the data list.
     labels = c("Bsl", "Vacc", "Vacc+RMD"),
     ## Change the clipping region.
     xpd = NA,
     ## Rotate the labels by 35 degrees.
     srt = 35,
     ## Adjust the labels to almost 100% right-justified.
     adj=0.4,
     ## Increase label size.
     cex = 1.4)

dev.off()


pdf("../RNASeq/MFuzzProfiles_new/GeneProfiles_Allgenes_cl7_cl5.pdf", width = 4, height = 4)
par(oma=c(0,0,0,1.2)) 
mfuzz.plot2(counts.matrix.s, cl = cl7, time.labels = c("Bsl", "Vacc", "Vacc+RMD"), mfrow = c(1,1), new.window = F, min.mem = 0.1, cex.axis = 1.4, x11=F, single =5, xaxt = "n", col.lab="transparent", col.main = "transparent")
axis(side = 1,labels = FALSE)
title(main = "Cluster 7\n (n = 747)", col.main ="black")
text(x = 1:3,
     ## Move labels to just below bottom of chart.
     y = par("usr")[3] - 0.45,
     ## Use names from the data list.
     labels = c("Bsl", "Vacc", "Vacc+RMD"),
     ## Change the clipping region.
     xpd = NA,
     ## Rotate the labels by 35 degrees.
     srt = 35,
     ## Adjust the labels to almost 100% right-justified.
     adj=0.4,
     ## Increase label size.
     cex = 1.4)

dev.off()


pdf("../RNASeq/MFuzzProfiles_new/GeneProfiles_Allgenes_cl7_cl6.pdf", width = 4, height = 4)
par(oma=c(0,0,0,1.2)) 
mfuzz.plot2(counts.matrix.s, cl = cl7, time.labels = c("Bsl", "Vacc", "Vacc+RMD"), mfrow = c(1,1), new.window = F, min.mem = 0.1, cex.axis = 1.4, x11=F, single =6, xaxt = "n", col.lab="transparent", col.main = "transparent")
axis(side = 1,labels = FALSE)
title(main = "Cluster 4\n (n = 359)", col.main ="black")
text(x = 1:3,
     ## Move labels to just below bottom of chart.
     y = par("usr")[3] - 0.45,
     ## Use names from the data list.
     labels = c("Bsl", "Vacc", "Vacc+RMD"),
     ## Change the clipping region.
     xpd = NA,
     ## Rotate the labels by 35 degrees.
     srt = 35,
     ## Adjust the labels to almost 100% right-justified.
     adj=0.4,
     ## Increase label size.
     cex = 1.4)

dev.off()



pdf("../RNASeq/MFuzzProfiles_new/GeneProfiles_Allgenes_cl7_cl7.pdf", width = 4, height = 4)
par(oma=c(0,0,0,1.2)) 
mfuzz.plot2(counts.matrix.s, cl = cl7, time.labels = c("Bsl", "Vacc", "Vacc+RMD"), mfrow = c(1,1), new.window = F, min.mem = 0.1, cex.axis = 1.4, x11=F, single =7, xaxt = "n", col.lab="transparent", col.main = "transparent")
axis(side = 1,labels = FALSE)
title(main = "Cluster 2\n (n = 423)", col.main ="black")
text(x = 1:3,
     ## Move labels to just below bottom of chart.
     y = par("usr")[3] - 0.45,
     ## Use names from the data list.
     labels = c("Bsl", "Vacc", "Vacc+RMD"),
     ## Change the clipping region.
     xpd = NA,
     ## Rotate the labels by 35 degrees.
     srt = 35,
     ## Adjust the labels to almost 100% right-justified.
     adj=0.4,
     ## Increase label size.
     cex = 1.4)

dev.off()
```




7. Make the overlap plot

PCA of the cluster centers. The lines indicate the overlap, the width makes reference to the stength of the overlap. 


```{r}

O.cl7 <- overlap(cl7)
overlap.plot(cl7, over = O.cl7, thres = 0)
pdf("../RNASeq/MFuzzProfiles_new/OverlapPlot_cl7.pdf")
overlap.plot(cl7, over = O.cl7, thres = 0)
dev.off()

```

8. Take the genes on each cluster and make the enrichment analysis

```{r}
membership <- cl7$membership
app <- apply(membership, 2, function(x) quantile(x, c(0.25, 0.50, 0.75, 0.80, 0.90, 0.95)))
median(app[4,])

write.table(app,"../RNASeq/MFuzzProfiles_new/membershipvalues.txt" )

genes_cluster <- acore(counts.matrix.s, cl7, min.acore = 0.1)
genes_cluster[[1]]

Genes_cl1 <- genes_cluster[[1]]
Genes_cl1$SYMBOL <- sapply(strsplit(rownames(Genes_cl1), ","), function(x)x[2])
Genes_cl1$Biotype <- sapply(strsplit(rownames(Genes_cl1), ","), function(x)x[3])
Genes_cl1$Transcript <- sapply(strsplit(rownames(Genes_cl1), ","), function(x)x[1])
Genes_cl1$EnsemblID <- sapply(strsplit(rownames(Genes_cl1), "[.]"), function(x)x[1])
write.table(Genes_cl1, file = "../RNASeq/MFuzzProfiles_new/Cl1Genes.txt", sep = "\t")


Genes_cl2 <- genes_cluster[[2]]
Genes_cl2$SYMBOL <- sapply(strsplit(rownames(Genes_cl2), ","), function(x)x[2])
Genes_cl2$Biotype <- sapply(strsplit(rownames(Genes_cl2), ","), function(x)x[3])
Genes_cl2$Transcript <- sapply(strsplit(rownames(Genes_cl2), ","), function(x)x[1])
Genes_cl2$EnsemblID <- sapply(strsplit(rownames(Genes_cl2), "[.]"), function(x)x[1])
write.table(Genes_cl2, file = "../RNASeq/MFuzzProfiles_new/cl2Genes.txt", sep = "\t")


Genes_cl3 <- genes_cluster[[3]]
Genes_cl3$SYMBOL <- sapply(strsplit(rownames(Genes_cl3), ","), function(x)x[2])
Genes_cl3$Biotype <- sapply(strsplit(rownames(Genes_cl3), ","), function(x)x[3])
Genes_cl3$Transcript <- sapply(strsplit(rownames(Genes_cl3), ","), function(x)x[1])
Genes_cl3$EnsemblID <- sapply(strsplit(rownames(Genes_cl3), "[.]"), function(x)x[1])
write.table(Genes_cl3, file = "../RNASeq/MFuzzProfiles_new/cl3Genes.txt", sep = "\t")


Genes_cl4 <- genes_cluster[[4]]
Genes_cl4$SYMBOL <- sapply(strsplit(rownames(Genes_cl4), ","), function(x)x[2])
Genes_cl4$Biotype <- sapply(strsplit(rownames(Genes_cl4), ","), function(x)x[3])
Genes_cl4$Transcript <- sapply(strsplit(rownames(Genes_cl4), ","), function(x)x[1])
Genes_cl4$EnsemblID <- sapply(strsplit(rownames(Genes_cl4), "[.]"), function(x)x[1])
write.table(Genes_cl4, file = "../RNASeq/MFuzzProfiles_new/cl4Genes.txt", sep = "\t")


Genes_cl5 <- genes_cluster[[5]]
Genes_cl5$SYMBOL <- sapply(strsplit(rownames(Genes_cl5), ","), function(x)x[2])
Genes_cl5$Biotype <- sapply(strsplit(rownames(Genes_cl5), ","), function(x)x[3])
Genes_cl5$Transcript <- sapply(strsplit(rownames(Genes_cl5), ","), function(x)x[1])
Genes_cl5$EnsemblID <- sapply(strsplit(rownames(Genes_cl5), "[.]"), function(x)x[1])
write.table(Genes_cl5, file = "../RNASeq/MFuzzProfiles_new/cl5Genes.txt", sep = "\t")



Genes_cl6 <- genes_cluster[[6]]
Genes_cl6$SYMBOL <- sapply(strsplit(rownames(Genes_cl6), ","), function(x)x[2])
Genes_cl6$Biotype <- sapply(strsplit(rownames(Genes_cl6), ","), function(x)x[3])
Genes_cl6$Transcript <- sapply(strsplit(rownames(Genes_cl6), ","), function(x)x[1])
Genes_cl6$EnsemblID <- sapply(strsplit(rownames(Genes_cl6), "[.]"), function(x)x[1])
write.table(Genes_cl6, file = "../RNASeq/MFuzzProfiles_new/cl6Genes.txt", sep = "\t")


Genes_cl7 <- genes_cluster[[7]]
Genes_cl7$SYMBOL <- sapply(strsplit(rownames(Genes_cl7), ","), function(x)x[2])
Genes_cl7$Biotype <- sapply(strsplit(rownames(Genes_cl7), ","), function(x)x[3])
Genes_cl7$Transcript <- sapply(strsplit(rownames(Genes_cl7), ","), function(x)x[1])
Genes_cl7$EnsemblID <- sapply(strsplit(rownames(Genes_cl7), "[.]"), function(x)x[1])
write.table(Genes_cl7, file = "../RNASeq/MFuzzProfiles_new/cl7Genes.txt", sep = "\t")
```


